homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -224,6 +251,46 @@ async def async_service_handler(service):
     async_add_entities(entities, update_before_add=True)
 
 
+class XiaomiGatewaySwitch(XiaomiGatewayDevice, SwitchEntity):
+    """Representation of a XiaomiGatewaySwitch."""
+
+    def __init__(self, coordinator, sub_device, entry, channel):
+        """Initialize the XiaomiSensor."""
+        super().__init__(coordinator, sub_device, entry)
+        self._channel = channel
+        self._data_key = f"status_ch{channel}"
+        self._unique_id = f"{sub_device.sid}-ch{channel}"
+        self._name = f"{sub_device.name} ch{channel} ({sub_device.sid})"
+
+    @property
+    def device_class(self):
+        """Return the device class of this entity."""
+        return DEVICE_CLASS_SWITCH
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._sub_device.status[self._data_key] == "on"
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the switch on."""
+        await self.hass.async_add_executor_job(
+            partial(self._sub_device.on, self._channel)
+        )
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the switch off."""
+        await self.hass.async_add_executor_job(
+            partial(self._sub_device.off, self._channel)
```

Comment:
We only need to use partial when we need to pass keyword arguments to `hass.async_add_executor_job`.

Reply:
thanks for the tip